client.go

```diff
@@ -155,87 +165,81 @@ func newPipe() pipe {
 	}
 }
 
-func (r remoteProcess) listen(ctx context.Context) {
-	defer r.conn.Close(websocket.StatusNormalClosure, "normal closure")
+func (r *remoteProcess) listen(ctx context.Context) {
 	defer close(r.done)
+	defer func() {
+		r.closeErr = r.conn.Close(websocket.StatusNormalClosure, "normal closure")
+	}()
+	defer func() {
+		r.stdoutErr = r.stdout.w.Close()
+		r.stderrErr = r.stderr.w.Close()
+	}()
```

Comment:
Is there a reason these are in different defer funcs?

Reply:
Historical.  I can combine